components/micro-gateway-cli/src/main/resources/templates/failoverEndpointReInit.mustache

```diff
@@ -1,10 +1,12 @@
-var err = trap
-{{#equals endpointUrlType "prod"}}{{#if hasProdEpConfig}}{{operationId}}{{else}}{{qualifiedServiceName}}{{/if}}_{{endpointUrlType}}.__init{{/equals}}
-{{#equals endpointUrlType "sand"}}{{#if hasSandEpConfig}}{{operationId}}{{else}}{{qualifiedServiceName}}{{/if}}_{{endpointUrlType}}.__init{{/equals}}
-({
-targets: [
-{{#endpoints}}    {url: {{#if isEtcdEnabled}}<string> getUrlOfEtcdKeyForReInit{{cut qualifiedServiceName " "}}("{{name}}_{{endpointUrlType}}_endpoint_{{@index}}","{{name}}_{{endpointUrlType}}_{{@index}}_etcdKey","{{etcdKey}}", "{{endpointUrl}}") {{else}} <string>gateway:retrieveConfig("{{name}}_{{endpointUrlType}}_endpoint_{{@index}}", "{{endpointUrl}}") {{/if}} {{>secureSocket}} }{{#unless @last}},
-{{/unless}}{{/endpoints}}
+http:FailoverClient|error reInitializedEndpoint = new({
+                                                  targets: [
+                                                  {{#endpoints}}    {url: {{#if isEtcdEnabled}}<string> getUrlOfEtcdKeyForReInit{{cut qualifiedServiceName " "}}("{{name}}_{{endpointUrlType}}_endpoint_{{@index}}","{{name}}_{{endpointUrlType}}_{{@index}}_etcdKey","{{etcdKey}}", "{{endpointUrl}}") {{else}} <string>gateway:retrieveConfig("{{name}}_{{endpointUrlType}}_endpoint_{{@index}}", "{{endpointUrl}}") {{/if}} {{>secureSocket}} }{{#unless @last}},
+                                                  {{/unless}}{{/endpoints}}
 
-],
-{{>http2}}{{>caching}}{{>basicAuth}} });
\ No newline at end of file
+                                                  ],
+                                                  {{>http2}}{{>caching}}{{>basicAuth}} });
+if(reInitializedEndpoint is http:FailoverClient) {
+{{#equals endpointUrlType "prod"}}{{#if hasProdEpConfig}}{{operationId}}{{else}}{{qualifiedServiceName}}{{/if}}_{{endpointUrlType}}{{/equals}}
+{{#equals endpointUrlType "sand"}}{{#if hasSandEpConfig}}{{operationId}}{{else}}{{qualifiedServiceName}}{{/if}}_{{endpointUrlType}}{{/equals}}
+= reInitializedEndpoint;
+}
```

Comment:
Why we need a another new line here

Reply:
This is a standard for file endings. Almost all unix terminal application requires this to function properly. Otherwise those programs don't process the last line of a file as a line.